File: ReclutaCVFrontend/src/communication/dtos/candidato.ts
import { VeredictoFinalCandidato } from "../enums/veredicto-final-candidato";

export interface CandidatoListable extends CandidatoConsultable {
    
}

export interface CandidatoInsertable extends CandidatoConsultable{
}

export interface CandidatoUpdatable extends CandidatoConsultable{
}

export interface CandidatoConsultable
{
    id: number | null;
    //Datos generales
    nombre: string;
    correo: string;
    telefono: string;
    fechaDeNacimiento: Date | null;
    estadoCivil: string;
    lugarNacimiento: string;
    generalesComentarios: string;
    fechaDeActualizacion: Date | null;

    // Experiencia laboral
    compañia: string;
    añosDeExperiencia: number | null;
    sueldoActual: number | null;
    motivoDeSeparacion: string;

    // Educación
    carrera: string;
    institucion: string;
    estatusAcademico: EstatusAcademico;
    cursos: string;
    certificaciones: string;
    competenciasOHabilidades: string;
    tecnologiasQueDomina: string;
    softwareQueDomina: string;

    // Inglés
    nivelDeIngles: string;
    nivelDeInglesHablado: number | null;
    nivelDeInglesEscrito: number | null;
    nivelDeInglesLectura: number | null;

    // Proceso de reclutamiento
    solicitudPersonalFolio: number | null;
    puestoClave: number | null;
    puestoNombre: string;
    proyectoNombre: string;
    fechaDeRecepcionCurriculum: Date | null;
    fechaDeContacto: Date | null;
    fechaPreentrevistaTelefonica: Date | null;
    fechaRecepcionSolicitudRegistro: Date | null;
    quienLoContacto: string;
    bolsa: BolsaTrabajo;
    bolsaOtra: string;
    rol: RolCandidato;
    rolOtro: string;
    expectativaEconomica: number | null;
    estatus: EstatusCandidato;
    reclutamientoComentarios: string;

    // examen psicometrico
    examenPsicometricoNombre: string;
    examenPsicometricoResultados: string;
    examenPsicometricoObservaciones: string;

    // examen de programación
    examenProgramacionFecha: Date | null;
    examenProgramacionIpComputadora: string;
    examenProgramacionId: string;
    examenProgramacionUmlCalificacion: number | null;
    examenProgramacionUmlTotalReactivos: number | null;
    examenProgramacionAdooCalificacion: number | null;
    examenProgramacionAdooTotalReactivos: number | null;
    examenProgramacionPooCalificacion: number | null;
    examenProgramacionPooTotalReactivos: number | null;
    examenProgramacionLogicaCalificacion: number | null;
    examenProgramacionLogicaTotalReactivos: number | null;
    examenProgramacionWebCalificacion: number | null;
    examenProgramacionWebTotalReactivos: number | null;
    examenProgramacionJavascriptCalificacion: number | null;
    examenProgramacionJavascriptTotalReactivos: number | null;
    examenProgramacionScrumCalificacion: number | null;
    examenProgramacionScrumTotalReactivos: number | null;
    examenProgramacionTecnologiaCalificacion: number | null;
    examenProgramacionTecnologiaTotalReactivos: number | null;
    examenProgramacionAciertos: number | null;
    examenProgramacionTotalReactivos: number | null;
    examenProgramacionRango: string;

    // examen de analista
    examenAnalistaFecha: Date | null;
    examenAnalistaIpComputadora: string;
    // Teórico
    examenAnalistaTeoricoId: string;
    examenAnalistaTeoricoAciertos: number | null;
    examenAnalistaTeoricoTotalReactivos: number | null;
    examenAnalistaTeoricoRango: string;
    // Práctico
    examenAnalistaPracticoId: string;
    examenAnalistaPracticoNumeroCaso: number | null;
    examenAnalistaPracticoAciertos: number | null;
    examenAnalistaPracticoTotalReactivos: number | null;
    examenAnalistaPracticoRango: string;

    // examen de ingeniero de pruebas
    examenIngenieroPruebasFecha: Date | null;
    examenIngenieroPruebasIpComputadora: string;
    // Teórico
    examenIngenieroPruebasTeoricoId: number | null;
    examenIngenieroPruebasTeoricoAciertos: number | null;
    examenIngenieroPruebasTeoricoTotalReactivos: number | null;
    examenIngenieroPruebasTeoricoRango: string;
    // Práctico
    examenIngenieroPruebasPracticoId: number | null;
    examenIngenieroPruebasPracticoCalificacion: number | null;
    examenIngenieroPruebasPracticoPuntos: number | null;
    examenIngenieroPruebasPracticoRango: string;
    // SQL
    examenIngenieroPruebasSqlTotalReactivos: number | null;
    examenIngenieroPruebasSqlCalificacion: number | null;

    // examen de administrador de proyectos
    examenAdministradorProyectoFecha: Date | null;
    examenAdministradorProyectoIpComputadora: string;
    examenAdministradorProyectoId: number | null;
    examenAdministradorProyectoAciertos: number | null;
    examenAdministradorProyectoTotalReactivos: number | null;
    examenAdministradorProyectoRango: string;

    // examen de soporte - BD
    examenPracticoSoporteBdFecha: Date | null;
    examenPracticoSoporteBdAciertos: number | null;
    examenPracticoSoporteBdTotalReactivos: number | null;
    examenPracticoSoporteBdRango: string;

    // Etapa de entrevistas
    entrevistaCapitalHumanoFecha: Date | null;
    entrevistaCapitalHumanoComentarios: string;
    entrevistaCoordinadorYEquipoTecnicoFecha: Date | null;
    entrevistaCoordinadorYEquipoTecnicoComentarios: string;
    entrevistaInglesFecha: Date | null;
    entrevistaInglesComentarios: string;
    entrevistaGerenteAreaFecha: Date | null;
    entrevistaGerenteAreaComentarios: string;
    veredictoFinal: VeredictoFinalCandidato;
    veredictoFinalNivelIdentificado: string;
    veredictoFinalComentarios: string;

    // Propuesta económica
    propuestaEconomicaFecha: Date | null;
    propuestaEconomicaEstatus: PropuestaEconomicaEstatus;
    propuestaEconomicaSueldo: number | null;
    propuestaEconomicaComentarios: string;

    // Ingreso
    ingresoFecha: Date | null;
    ingresoTipoContrato: string;
    ingresoVencimientoContratoDeterminado: Date | null;
    ingresoObservaciones: string;
}

File: ReclutaCVApi/Dtos/Candidato.cs
using ReclutaCVData.Entidades;
using System;

namespace ReclutaCVApi.Dtos
{
    public class CandidatoInsertable : CandidatoConsultable
    {
    }

    public class CandidatoUpdatable : CandidatoConsultable
    {
    }

    public class CandidatoConsultable
    {
        public int? Id { get; set; }
        //Datos generales
        public string Nombre { get; set; }
        public string Correo { get; set; }
        public string Telefono { get; set; }
        public DateTime? FechaDeNacimiento { get; set; }
        public string EstadoCivil { get; set; }
        public string LugarNacimiento { get; set; }
        public string GeneralesComentarios { get; set; }
        public DateTime? FechaDeActualizacion { get; set; }

        // Experiencia laboral
        public string Compañia { get; set; }
        public decimal? AñosDeExperiencia { get; set; }
        public decimal? SueldoActual { get; set; }
        public string MotivoDeSeparacion { get; set; }

        // Educación
        public string Carrera { get; set; }
        public string Institucion { get; set; }
        public EstatusAcademico EstatusAcademico { get; set; } = EstatusAcademico.NoEspecificado;
        public string Cursos { get; set; }
        public string Certificaciones { get; set; }
        public string CompetenciasOHabilidades { get; set; }
        public string TecnologiasQueDomina { get; set; }
        public string SoftwareQueDomina { get; set; }

        // Inglés
        public string NivelDeIngles { get; set; }
        public decimal? NivelDeInglesHablado { get; set; }
        public decimal? NivelDeInglesEscrito { get; set; }
        public decimal? NivelDeInglesLectura { get; set; }

        // Proceso de reclutamiento
        public int? SolicitudPersonalFolio { get; set; }
        public int? PuestoClave { get; set; }
        public string PuestoNombre { get; set; }
        public string ProyectoNombre { get; set; }
        public DateTime? FechaDeRecepcionCurriculum { get; set; }
        public DateTime? FechaDeContacto { get; set; }
        public DateTime? FechaPreentrevistaTelefonica { get; set; }
        public DateTime? FechaRecepcionSolicitudRegistro { get; set; }
        public string QuienLoContacto { get; set; }
        public BolsaTrabajo Bolsa { get; set; } = BolsaTrabajo.NoEspecificada;
        public string BolsaOtra { get; set; }
        public RolCandidato Rol { get; set; } = RolCandidato.IngenieroDeSoftware;
        public string RolOtro { get; set; }
        public decimal? ExpectativaEconomica { get; set; }
        public EstatusCandidato Estatus { get; set; } = EstatusCandidato.NoEspecificado;
        public string ReclutamientoComentarios { get; set; }

        // Examen psicometrico
        public string ExamenPsicometricoNombre { get; set; }
        public string ExamenPsicometricoResultados { get; set; }
        public string ExamenPsicometricoObservaciones { get; set; }

        // Examen de programación
        public DateTime? ExamenProgramacionFecha { get; set; }
        public string ExamenProgramacionIpComputadora { get; set; }
        public string ExamenProgramacionId { get; set; }
        public int? ExamenProgramacionUmlCalificacion { get; set; }
        public int? ExamenProgramacionUmlTotalReactivos { get; set; }
        public int? ExamenProgramacionAdooCalificacion { get; set; }
        public int? ExamenProgramacionAdooTotalReactivos { get; set; }
        public int? ExamenProgramacionPooCalificacion { get; set; }
        public int? ExamenProgramacionPooTotalReactivos { get; set; }
        public int? ExamenProgramacionLogicaCalificacion { get; set; }
        public int? ExamenProgramacionLogicaTotalReactivos { get; set; }
        public int? ExamenProgramacionWebCalificacion { get; set; }
        public int? ExamenProgramacionWebTotalReactivos { get; set; }
        public int? ExamenProgramacionJavascriptCalificacion { get; set; }
        public int? ExamenProgramacionJavascriptTotalReactivos { get; set; }
        public int? ExamenProgramacionScrumCalificacion { get; set; }
        public int? ExamenProgramacionScrumTotalReactivos { get; set; }
        public int? ExamenProgramacionTecnologiaCalificacion { get; set; }
        public int? ExamenProgramacionTecnologiaTotalReactivos { get; set; }
        public int? ExamenProgramacionAciertos { get; set; }
        public int? ExamenProgramacionTotalReactivos { get; set; }
        public string ExamenProgramacionRango { get; set; }

        // Examen de analista
        public DateTime? ExamenAnalistaFecha { get; set; }
        public string ExamenAnalistaIpComputadora { get; set; }
        //// Teórico
        public string ExamenAnalistaTeoricoId { get; set; }
        public int? ExamenAnalistaTeoricoAciertos { get; set; }
        public int? ExamenAnalistaTeoricoTotalReactivos { get; set; }
        public string ExamenAnalistaTeoricoRango { get; set; }
        //// Práctico
        public string ExamenAnalistaPracticoId { get; set; }
        public int? ExamenAnalistaPracticoNumeroCaso { get; set; }
        public int? ExamenAnalistaPracticoAciertos { get; set; }
        public int? ExamenAnalistaPracticoTotalReactivos { get; set; }
        public string ExamenAnalistaPracticoRango { get; set; }

        // Examen de ingeniero de pruebas
        public DateTime? ExamenIngenieroPruebasFecha { get; set; }
        public string ExamenIngenieroPruebasIpComputadora { get; set; }
        //// Teórico
        public int? ExamenIngenieroPruebasTeoricoId { get; set; }
        public int? ExamenIngenieroPruebasTeoricoAciertos { get; set; }
        public int? ExamenIngenieroPruebasTeoricoTotalReactivos { get; set; }
        public string ExamenIngenieroPruebasTeoricoRango { get; set; }
        //// Práctico
        public int? ExamenIngenieroPruebasPracticoId { get; set; }
        public decimal? ExamenIngenieroPruebasPracticoCalificacion { get; set; }
        public int? ExamenIngenieroPruebasPracticoPuntos { get; set; }
        public string ExamenIngenieroPruebasPracticoRango { get; set; }
        //// SQL
        public int? ExamenIngenieroPruebasSqlTotalReactivos { get; set; }
        public decimal? ExamenIngenieroPruebasSqlCalificacion { get; set; }

        // Examen de administrador de proyectos
        public DateTime? ExamenAdministradorProyectoFecha { get; set; }
        public string ExamenAdministradorProyectoIpComputadora { get; set; }
        public int? ExamenAdministradorProyectoId { get; set; }
        public int? ExamenAdministradorProyectoAciertos { get; set; }
        public int? ExamenAdministradorProyectoTotalReactivos { get; set; }
        public string ExamenAdministradorProyectoRango { get; set; }

        // Examen de soporte - BD
        public DateTime? ExamenPracticoSoporteBdFecha { get; set; }
        public int? ExamenPracticoSoporteBdAciertos { get; set; }
        public int? ExamenPracticoSoporteBdTotalReactivos { get; set; }
        public string ExamenPracticoSoporteBdRango { get; set; }

        // Etapa de entrevistas
        public DateTime? EntrevistaCapitalHumanoFecha { get; set; }
        public string EntrevistaCapitalHumanoComentarios { get; set; }
        public DateTime? EntrevistaCoordinadorYEquipoTecnicoFecha { get; set; }
        public string EntrevistaCoordinadorYEquipoTecnicoComentarios { get; set; }
        public DateTime? EntrevistaInglesFecha { get; set; }
        public string EntrevistaInglesComentarios { get; set; }
        public DateTime? EntrevistaGerenteAreaFecha { get; set; }
        public string EntrevistaGerenteAreaComentarios { get; set; }
        public VeredictoFinalCandidato VeredictoFinal { get; set; } = VeredictoFinalCandidato.EnEspera;
        public string VeredictoFinalNivelIdentificado { get; set; }
        public string VeredictoFinalComentarios { get; set; }

        // Propuesta económica
        public DateTime? PropuestaEconomicaFecha { get; set; }
        public PropuestaEconomicaEstatus PropuestaEconomicaEstatus { get; set; } = PropuestaEconomicaEstatus.SinPropuesta;
        public decimal? PropuestaEconomicaSueldo { get; set; }
        public string PropuestaEconomicaComentarios { get; set; }

        // Ingreso
        public DateTime? IngresoFecha { get; set; }
        public string IngresoTipoContrato { get; set; }
        public DateTime? IngresoVencimientoContratoDeterminado { get; set; }
        public string IngresoObservaciones { get; set; }
    }
}